internal/websocket/ws.go

```diff
@@ -169,6 +172,12 @@ func (ws *Websocket) Stop() {
 	ws.wg.Wait()
 }
 
+// AddMessageListener adds a message listener to the WS connection.
+// The listener will be called for every message received from the WS.
+func (ws *Websocket) AddMessageListener(listener event.MsgListener, cancel ...*event.CancelFunc) event.CancelFunc {
+	return event.CancelFunc(ws.msgListeners.Add(listener))
+}
+
```

Comment:
We needed a way to add listeners after the fact, i.e. adding a ws - to - wrphandler adapter